Tests/LibCore/TestLibCoreStream.cpp

```diff
@@ -344,21 +293,15 @@ TEST_CASE(local_socket_read)
         [](auto&) {
             Core::EventLoop event_loop;
 
-            auto maybe_client_socket = Core::LocalSocket::connect("/tmp/test-socket");
-            EXPECT(!maybe_client_socket.is_error());
-            auto client_socket = maybe_client_socket.release_value();
+            auto client_socket = MUST(Core::LocalSocket::connect("/tmp/test-socket"));
 
             EXPECT(client_socket->is_open());
 
             EXPECT(client_socket->can_read_without_blocking(100).release_value());
             EXPECT_EQ(client_socket->pending_bytes().release_value(), sent_data.length());
 
-            auto maybe_receive_buffer = ByteBuffer::create_uninitialized(64);
-            EXPECT(!maybe_receive_buffer.is_error());
-            auto receive_buffer = maybe_receive_buffer.release_value();
-            auto maybe_read_bytes = client_socket->read_some(receive_buffer);
-            EXPECT(!maybe_read_bytes.is_error());
-            auto read_bytes = maybe_read_bytes.release_value();
+            auto receive_buffer = MUST(ByteBuffer::create_uninitialized(64));
+            auto read_bytes = MUST(client_socket->read_some(receive_buffer));
```

Comment:
Lines 303, 304 and 296 here are MUST() instead of TRY_OR_FAIL. Is that b/c of the different thread context?

Reply:
`TRY_OR_FAIL` can not be used inside functions. You would need to change the function to return `ErrorOr<T>`, use `TRY` and `TRY_OR_FAIL` the function call.

Reply:
Jup, exactly what Fabian said. I used `TRY_OR_FAIL` in all possible contexts, and the remaining `MUST`s cannot be replaced by `TRY_OR_FAIL` without also changing the code more invasively (i.e. changing return types).

Reply:
On a side note: This code runs in a different thread, and already calls `EXPECT`, which in turn sets `TestSuite::m_current_test_case_passed`, which is a simple `bool`. In other words, that's a data race, or at least feels dangerously close to one.